src/daft-core/src/array/ops/list.rs

```diff
@@ -302,4 +479,133 @@ impl FixedSizeListArray {
+                let arr = casted.i64()?;
+
+                self.agg_data_array(arr, &sum_primitive)
+            }
+            UInt8 | UInt16 | UInt32 | UInt64 => {
+                let casted = self.flat_child.cast(&UInt64)?;
+                let arr = casted.u64()?;
+
+                self.agg_data_array(arr, &sum_primitive)
+            }
+            Float32 => self.agg_data_array(self.flat_child.f32()?, &sum_primitive),
+            Float64 => self.agg_data_array(self.flat_child.f64()?, &sum_primitive),
+            other => Err(DaftError::TypeError(format!(
+                "Sum not implemented for {}",
+                other
+            ))),
+        }
+    }
+
+    pub fn mean(&self) -> DaftResult<Float64Array> {
+        use crate::datatypes::DataType::*;
+
+        match self.flat_child.data_type() {
+            Int8 | Int16 | Int32 | Int64 | UInt8 | UInt16 | UInt32 | UInt64 | Float32 | Float64 => {
+                let counts = self.count(CountMode::Valid)?;
+                let sum_series = self.sum()?.cast(&Float64)?;
+                let sums = sum_series.f64()?;
+
+                let means = counts
+                    .into_iter()
+                    .zip(sums)
+                    .map(|(count, sum)| match (count, sum) {
+                        (Some(count), Some(sum)) if *count != 0 => Some(sum / (*count as f64)),
+                        _ => None,
+                    });
+
+                let arr = Box::new(arrow2::array::PrimitiveArray::from_trusted_len_iter(means));
+
+                Ok(Float64Array::from((self.name(), arr)))
+            }
+            other => Err(DaftError::TypeError(format!(
+                "Mean not implemented for {}",
+                other
+            ))),
+        }
+    }
+
+    pub fn min(&self) -> DaftResult<Series> {
+        use crate::datatypes::DataType::*;
+        use arrow2::compute::aggregate::min_primitive;
+
+        match self.flat_child.data_type() {
+            Int8 => self.agg_data_array(self.flat_child.i8()?, &min_primitive),
+            Int16 => self.agg_data_array(self.flat_child.i16()?, &min_primitive),
+            Int32 => self.agg_data_array(self.flat_child.i32()?, &min_primitive),
+            Int64 => self.agg_data_array(self.flat_child.i64()?, &min_primitive),
+            UInt8 => self.agg_data_array(self.flat_child.u8()?, &min_primitive),
+            UInt16 => self.agg_data_array(self.flat_child.u16()?, &min_primitive),
+            UInt32 => self.agg_data_array(self.flat_child.u32()?, &min_primitive),
+            UInt64 => self.agg_data_array(self.flat_child.u64()?, &min_primitive),
+            Float32 => self.agg_data_array(self.flat_child.f32()?, &min_primitive),
+            Float64 => self.agg_data_array(self.flat_child.f64()?, &min_primitive),
+            other => Err(DaftError::TypeError(format!(
+                "Min not implemented for {}",
+                other
+            ))),
+        }
+    }
```

Comment:
Any suggestions for how to write this better would be appreciated. While you usually can solve this by doing like `self.inner.list_min()` in `Series::list_min` and implement it for each SeriesLike, this doesn't work the same way since it's the child type that varies, whereas the parent Series will always be a ListArray

Reply:
I'm not entirely sure if this is what you meant but I tried implementing something along those lines, but only for min/max since there is no code that we can share for sum and count between the two list types.

It still has the long set of matches for each aggregation, but at least it's done once per function and not also for each list type too